images/update_kcp_stable/bin/update.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+declare -r TARGET_BRANCH="main"
+declare -r gitlab_ci="./.gitlab-ci.yml"
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$0")" >/dev/null
+  pwd
+)"
+
+# shellcheck source=images/update_kcp_stable/bin/gitlab_open_mr.sh
+source "$SCRIPT_DIR/gitlab_open_mr.sh"
+
+usage() {
+
+  printf "
+Usage:
+    %s [options]
+
+Fetch the latest commit from Pipeline Service Github repo
+and update the image tags in the Gitlab repo where the config lives.
+
+Optional arguments:
+    -scm, --source-control-management SCM
+        Source Code Management tools like github, gitlab, bitbucket etc.
+        Default: gitlab
```

Comment:
Not sure it makes sense to have a default value for this.
Also, I would assume that by looking at env variables in the container, the script should be able to find the SCM on its own (e.g. if `GITLAB_PROJECT_ID` exists, then we're in GitLab).

Reply:
I think its still good to provide an option for a user to mention this. Because, if we need to implement auto-detection of SCM, then we would need to do that for a lot of providers.
Default value -> The reason to mention the default value is because our project is using Gitlab. If we don't provide a default, then developers using this would need to  use the flag every time they use the script.